<!DOCTYPE HTML>
<html lang="en" class="sidebar-visible no-js light">
    <head>
        <!-- Book generated using mdBook -->
        <meta charset="UTF-8">
        <title>gget</title>
        <meta name="robots" content="noindex" />
        <!-- Custom HTML head -->
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-ZM4N3XBEW7"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-ZM4N3XBEW7');
</script>
        <meta content="text/html; charset=utf-8" http-equiv="Content-Type">
        <meta name="description" content="">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="theme-color" content="#ffffff" />

        <link rel="icon" href="favicon.svg">
        <link rel="shortcut icon" href="favicon.png">
        <link rel="stylesheet" href="css/variables.css">
        <link rel="stylesheet" href="css/general.css">
        <link rel="stylesheet" href="css/chrome.css">
        <link rel="stylesheet" href="css/print.css" media="print">
        <!-- Fonts -->
        <link rel="stylesheet" href="FontAwesome/css/font-awesome.css">
        <link rel="stylesheet" href="fonts/fonts.css">
        <!-- Highlight.js Stylesheets -->
        <link rel="stylesheet" href="highlight.css">
        <link rel="stylesheet" href="tomorrow-night.css">
        <link rel="stylesheet" href="ayu-highlight.css">

        <!-- Custom theme stylesheets -->
    </head>
    <body>
        <!-- Provide site root to javascript -->
        <script type="text/javascript">
            var path_to_root = "";
            var default_theme = window.matchMedia("(prefers-color-scheme: dark)").matches ? "navy" : "light";
        </script>

        <!-- Work around some values being stored in localStorage wrapped in quotes -->
        <script type="text/javascript">
            try {
                var theme = localStorage.getItem('mdbook-theme');
                var sidebar = localStorage.getItem('mdbook-sidebar');

                if (theme.startsWith('"') && theme.endsWith('"')) {
                    localStorage.setItem('mdbook-theme', theme.slice(1, theme.length - 1));
                }

                if (sidebar.startsWith('"') && sidebar.endsWith('"')) {
                    localStorage.setItem('mdbook-sidebar', sidebar.slice(1, sidebar.length - 1));
                }
            } catch (e) { }
        </script>

        <!-- Set the theme before any content is loaded, prevents flash -->
        <script type="text/javascript">
            var theme;
            try { theme = localStorage.getItem('mdbook-theme'); } catch(e) { }
            if (theme === null || theme === undefined) { theme = default_theme; }
            var html = document.querySelector('html');
            html.classList.remove('no-js')
            html.classList.remove('light')
            html.classList.add(theme);
            html.classList.add('js');
        </script>

        <!-- Hide / unhide sidebar before it is displayed -->
        <script type="text/javascript">
            var html = document.querySelector('html');
            var sidebar = 'hidden';
            if (document.body.clientWidth >= 1080) {
                try { sidebar = localStorage.getItem('mdbook-sidebar'); } catch(e) { }
                sidebar = sidebar || 'visible';
            }
            html.classList.remove('sidebar-visible');
            html.classList.add("sidebar-" + sidebar);
        </script>

        <nav id="sidebar" class="sidebar" aria-label="Table of contents">
            <div class="sidebar-scrollbox">
                <ol class="chapter"><li class="chapter-item expanded affix "><a href="introduction.html">Introduction</a></li><li class="chapter-item expanded affix "><a href="updates.html">What's New?</a></li><li class="chapter-item expanded affix "><li class="part-title">User Guide</li><li class="chapter-item expanded "><a href="installation.html"><strong aria-hidden="true">1.</strong> Installation</a></li><li class="chapter-item expanded "><a href="modules.html"><strong aria-hidden="true">2.</strong> Modules</a></li><li class="chapter-item expanded "><a href="quick_start_guide.html"><strong aria-hidden="true">3.</strong> Quick Start Guide</a></li><li class="chapter-item expanded affix "><li class="part-title">Manual</li><li class="chapter-item expanded "><a href="ref.html"><strong aria-hidden="true">4.</strong> gget ref</a></li><li class="chapter-item expanded "><a href="search.html"><strong aria-hidden="true">5.</strong> gget search</a></li><li class="chapter-item expanded "><a href="info.html"><strong aria-hidden="true">6.</strong> gget info</a></li><li class="chapter-item expanded "><a href="seq.html"><strong aria-hidden="true">7.</strong> gget seq</a></li><li class="chapter-item expanded "><a href="blast.html"><strong aria-hidden="true">8.</strong> gget blast</a></li><li class="chapter-item expanded "><a href="blat.html"><strong aria-hidden="true">9.</strong> gget blat</a></li><li class="chapter-item expanded "><a href="muscle.html"><strong aria-hidden="true">10.</strong> gget muscle</a></li><li class="chapter-item expanded "><a href="enrichr.html"><strong aria-hidden="true">11.</strong> gget enrichr</a></li><li class="chapter-item expanded "><a href="archs4.html"><strong aria-hidden="true">12.</strong> gget archs4</a></li><li class="chapter-item expanded "><a href="pdb.html"><strong aria-hidden="true">13.</strong> gget pdb</a></li><li class="chapter-item expanded "><a href="alphafold.html"><strong aria-hidden="true">14.</strong> gget alphafold</a></li><li class="chapter-item expanded "><a href="setup.html"><strong aria-hidden="true">15.</strong> gget setup</a></li><li class="spacer"></li><li class="chapter-item expanded affix "><a href="cite.html">Terms of Use</a></li></ol>
            </div>
            <div id="sidebar-resize-handle" class="sidebar-resize-handle"></div>
        </nav>

        <div id="page-wrapper" class="page-wrapper">

            <div class="page">
                <div id="menu-bar-hover-placeholder"></div>
                <div id="menu-bar" class="menu-bar sticky bordered">
                    <div class="left-buttons">
                        <button id="sidebar-toggle" class="icon-button" type="button" title="Toggle Table of Contents" aria-label="Toggle Table of Contents" aria-controls="sidebar">
                            <i class="fa fa-bars"></i>
                        </button>
                        <button id="theme-toggle" class="icon-button" type="button" title="Change theme" aria-label="Change theme" aria-haspopup="true" aria-expanded="false" aria-controls="theme-list">
                            <i class="fa fa-paint-brush"></i>
                        </button>
                        <ul id="theme-list" class="theme-popup" aria-label="Themes" role="menu">
                            <li role="none"><button role="menuitem" class="theme" id="light">Light (default)</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="rust">Rust</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="coal">Coal</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="navy">Navy</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="ayu">Ayu</button></li>
                        </ul>
                        <button id="search-toggle" class="icon-button" type="button" title="Search. (Shortkey: s)" aria-label="Toggle Searchbar" aria-expanded="false" aria-keyshortcuts="S" aria-controls="searchbar">
                            <i class="fa fa-search"></i>
                        </button>
                    </div>

                    <h1 class="menu-title">gget</h1>

                    <div class="right-buttons">
                        <a href="print.html" title="Print this book" aria-label="Print this book">
                            <i id="print-button" class="fa fa-print"></i>
                        </a>
                    </div>
                </div>

                <div id="search-wrapper" class="hidden">
                    <form id="searchbar-outer" class="searchbar-outer">
                        <input type="search" id="searchbar" name="searchbar" placeholder="Search this book ..." aria-controls="searchresults-outer" aria-describedby="searchresults-header">
                    </form>
                    <div id="searchresults-outer" class="searchresults-outer hidden">
                        <div id="searchresults-header" class="searchresults-header"></div>
                        <ul id="searchresults">
                        </ul>
                    </div>
                </div>
                <!-- Apply ARIA attributes after the sidebar and the sidebar toggle button are added to the DOM -->
                <script type="text/javascript">
                    document.getElementById('sidebar-toggle').setAttribute('aria-expanded', sidebar === 'visible');
                    document.getElementById('sidebar').setAttribute('aria-hidden', sidebar !== 'visible');
                    Array.from(document.querySelectorAll('#sidebar a')).forEach(function(link) {
                        link.setAttribute('tabIndex', sidebar === 'visible' ? 0 : -1);
                    });
                </script>

                <div id="content" class="content">
                    <main>
                        <p><a href="https://pypi.org/project/gget"><img src="https://img.shields.io/pypi/v/gget" alt="pypi version" /></a>
<a href="https://anaconda.org/bioconda/gget"><img src="https://anaconda.org/bioconda/gget/badges/version.svg" alt="image" /></a>
<a href="https://pepy.tech/project/gget"><img src="https://static.pepy.tech/personalized-badge/gget?period=total&amp;units=international_system&amp;left_color=grey&amp;right_color=brightgreen&amp;left_text=Downloads" alt="Downloads" /></a>
<a href="LICENSE"><img src="https://img.shields.io/pypi/l/gget" alt="license" /></a>
<img src="https://github.com/pachterlab/gget/workflows/CI/badge.svg" alt="status" />
<img src="https://github.com/pachterlab/gget/workflows/CI_alphafold/badge.svg?branch=test_gget_alphafold" alt="status" />
<a href="https://github.com/pachterlab/gget/"><img src="https://img.shields.io/github/stars/pachterlab/gget.svg?style=social" alt="Star on GitHub" /></a></p>
<p><code>gget</code> is a free, open-source command-line tool and Python package that enables efficient querying of genomic databases. <code>gget</code>  consists of a collection of separate but interoperable modules, each designed to facilitate one type of database querying in a single line of code.</p>
<p><img src="https://github.com/pachterlab/gget/blob/main/figures/gget_overview.png?raw=true" alt="alt text" /></p>
<p>If you use <code>gget</code> in a publication, please <a href="cite.html">cite*</a>:</p>
<pre><code>Luebbert, L., &amp; Pachter, L. (2023). Efficient querying of genomic reference databases with gget. Bioinformatics. https://doi.org/10.1093/bioinformatics/btac836
</code></pre>
<p>Read the article here: <a href="https://doi.org/10.1093/bioinformatics/btac836">https://doi.org/10.1093/bioinformatics/btac836</a></p>
<div style="break-before: page; page-break-before: always;"></div><h2 id="-whats-new"><a class="header" href="#-whats-new">✨ What's new</a></h2>
<p><strong>Version ≥ 0.27.2:</strong></p>
<ul>
<li>Updated <a href="./alphafold.html"><code>gget alphafold</code></a> to <a href="https://github.com/deepmind/alphafold/releases/tag/v2.3.0">DeepMind's AlphaFold v2.3.0</a> (including new arguments <code>multimer_for_monomer</code> and <code>multimer_recycles</code>)</li>
</ul>
<p><strong>Version ≥ 0.27.0:</strong></p>
<ul>
<li>Updated <a href="./alphafold.html"><code>gget alphafold</code></a> to match recent changes by DeepMind</li>
<li>Updated version number to match gget's creator's age following a long-standing Pachter lab tradition</li>
</ul>
<p><strong>Version ≥ 0.3.13:</strong></p>
<ul>
<li>Reduced runtime for <a href="./enrichr.html"><code>gget enrichr</code></a> and <a href="./archs4.html"><code>gget archs4</code></a> when used with Ensembl IDs</li>
</ul>
<p><strong>Version ≥ 0.3.12:</strong></p>
<ul>
<li><a href="./info.html"><code>gget info</code></a> now also returns subcellular localisation data from UniProt</li>
<li>New <a href="./info.html"><code>gget info</code></a> flag <code>ensembl_only</code> returns only Ensembl results</li>
<li>Reduced runtime for <a href="./info.html"><code>gget info</code></a> and <a href="./seq.html"><code>gget seq</code></a></li>
</ul>
<p><strong>Version ≥ 0.3.11: <a href="./pdb.html"><code>gget pdb</code></a></strong></p>
<p><strong>NOTE:</strong> The <a href="http://ftp.ensembl.org/pub/">Ensembl FTP site</a> changed its structure on August 8, 2022. Please upgrade to <code>gget</code> version ≥ 0.3.7 if you use <code>gget ref</code>.</p>
<p><strong>Version ≥ 0.3.0: <a href="./alphafold.html"><code>gget alphafold</code></a></strong></p>
<p><strong>NOTE:</strong> <a href="https://www.uniprot.org/">UniProt</a> changed the structure of their API on June 28, 2022. Please upgrade to <code>gget</code> version ≥ 0.2.5 if you use any of the modules querying data from UniProt (<code>gget info</code> and <code>gget seq</code>).</p>
<p><strong>Version ≥ 0.2.0:</strong></p>
<ul>
<li>JSON is now the default output format for the command-line interface for modules that previously returned data frame (CSV) format by default (the output can be converted to data frame/CSV using flag <code>[-csv][--csv]</code>). Data frame/CSV remains the default output for Jupyter Lab / Google Colab (and can be converted to JSON with <code>json=True</code>).</li>
<li>For all modules, the first required argument was converted to a positional argument and should not be named anymore in the command-line, e.g. <code>gget ref -s human</code> → <code>gget ref human</code>.</li>
<li><code>gget info</code>: <code>[--expand]</code> is deprecated. The module will now always return all of the available information.</li>
<li>Slight changes to the output returned by <code>gget info</code>, including the return of versioned Ensembl IDs.</li>
<li><code>gget info</code> and <code>gget seq</code> now support 🪱 WormBase and 🪰 FlyBase IDs.</li>
<li><code>gget archs4</code> and <code>gget enrichr</code> now also take Ensembl IDs as input with added flag <code>[-e][--ensembl]</code> (<code>ensembl=True</code> in Jupyter Lab / Google Colab).</li>
<li><code>gget seq</code> argument <code>seqtype</code> was replaced by flag <code>[-t][--translate]</code> (<code>translate=True/False</code> in Jupyter Lab / Google Colab) which will return either nucleotide (<code>False</code>) or amino acid (<code>True</code>) sequences.</li>
<li><code>gget search</code> argument <code>seqtype</code> was renamed to <code>id_type</code> for clarity (still taking the same arguments 'gene' or 'transcript').</li>
<li>Version ≥ 0.2.6: <code>gget ref</code> supports plant genomes! 🌱</li>
</ul>
<div style="break-before: page; page-break-before: always;"></div><p><a href="https://pypi.org/project/gget"><img src="https://img.shields.io/pypi/v/gget" alt="pypi version" /></a>
<a href="https://anaconda.org/bioconda/gget"><img src="https://anaconda.org/bioconda/gget/badges/version.svg" alt="image" /></a></p>
<h2 id="installation"><a class="header" href="#installation">Installation</a></h2>
<pre><code class="language-bash">pip install --upgrade gget
</code></pre>
<p>Alternative:</p>
<pre><code class="language-bash">conda install -c bioconda gget
</code></pre>
<p>For use in Jupyter Lab / Google Colab:</p>
<pre><code class="language-python">import gget
</code></pre>
<div style="break-before: page; page-break-before: always;"></div><p><code>gget</code> currently consists of the following modules:</p>
<ul>
<li>
<p><a href="ref.html"><code>gget ref</code></a><br />
Fetch File Transfer Protocols (FTPs) and metadata for reference genomes and annotations from <a href="https://www.ensembl.org/">Ensembl</a> by species.</p>
</li>
<li>
<p><a href="search.html"><code>gget search</code></a><br />
Fetch genes and transcripts from <a href="https://www.ensembl.org/">Ensembl</a> using free-form search terms.</p>
</li>
<li>
<p><a href="info.html"><code>gget info</code></a><br />
Fetch extensive gene and transcript metadata from <a href="https://www.ensembl.org/">Ensembl</a>, <a href="https://www.uniprot.org/">UniProt</a>, and <a href="https://www.ncbi.nlm.nih.gov/">NCBI</a> using Ensembl IDs.</p>
</li>
<li>
<p><a href="seq.html"><code>gget seq</code></a><br />
Fetch nucleotide or amino acid sequences of genes or transcripts from <a href="https://www.ensembl.org/">Ensembl</a> or <a href="https://www.uniprot.org/">UniProt</a>, respectively.</p>
</li>
<li>
<p><a href="blast.html"><code>gget blast</code></a><br />
BLAST a nucleotide or amino acid sequence to any <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a> database.</p>
</li>
<li>
<p><a href="blat.html"><code>gget blat</code></a><br />
Find the genomic location of a nucleotide or amino acid sequence using <a href="https://genome.ucsc.edu/cgi-bin/hgBlat">BLAT</a>.</p>
</li>
<li>
<p><a href="muscle.html"><code>gget muscle</code></a><br />
Align multiple nucleotide or amino acid sequences to each other using <a href="https://www.drive5.com/muscle/">Muscle5</a>.</p>
</li>
<li>
<p><a href="enrichr.html"><code>gget enrichr</code></a><br />
Perform an enrichment analysis on a list of genes using <a href="https://maayanlab.cloud/Enrichr/">Enrichr</a>.</p>
</li>
<li>
<p><a href="archs4.html"><code>gget archs4</code></a><br />
Find the most correlated genes to a gene of interest or find the gene's tissue expression atlas using <a href="https://maayanlab.cloud/archs4/">ARCHS4</a>.</p>
</li>
<li>
<p><a href="pdb.html"><code>gget pdb</code></a><br />
Get the structure and metadata of a protein from the <a href="https://www.rcsb.org/">RCSB Protein Data Bank</a>. </p>
</li>
<li>
<p><a href="alphafold.html"><code>gget alphafold</code></a><br />
Predict the 3D structure of a protein from its amino acid sequence using a simplified version of <a href="https://www.deepmind.com/">DeepMind</a>’s <a href="https://github.com/deepmind/alphafold">AlphaFold2</a>.</p>
</li>
</ul>
<div style="break-before: page; page-break-before: always;"></div><h2 id="-quick-start-guide"><a class="header" href="#-quick-start-guide">🪄 Quick start guide</a></h2>
<p>Command line:</p>
<pre><code class="language-bash"># Fetch all Homo sapiens reference and annotation FTPs from the latest Ensembl release
$ gget ref homo_sapiens

# Get Ensembl IDs of human genes with &quot;ace2&quot; or &quot;angiotensin converting enzyme 2&quot; in their name/description
$ gget search -s homo_sapiens 'ace2' 'angiotensin converting enzyme 2'

# Look up gene ENSG00000130234 (ACE2) and its transcript ENST00000252519
$ gget info ENSG00000130234 ENST00000252519

# Fetch the amino acid sequence of the canonical transcript of gene ENSG00000130234
$ gget seq --translate ENSG00000130234

# Quickly find the genomic location of (the start of) that amino acid sequence
$ gget blat MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS

# BLAST (the start of) that amino acid sequence
$ gget blast MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS

# Align nucleotide or amino acid sequences stored in a FASTA file
$ gget muscle path/to/file.fa

# Use Enrichr for an ontology analysis of a list of genes
$ gget enrichr -db ontology ACE2 AGT AGTR1 ACE AGTRAP AGTR2 ACE3P

# Get the human tissue expression of gene ACE2
$ gget archs4 -w tissue ACE2

# Get the protein structure (in PDB format) of ACE2 as stored in the Protein Data Bank (PDB ID returned by gget info)
$ gget pdb 1R42 -o 1R42.pdb

# Predict the protein structure of GFP from its amino acid sequence
$ gget setup alphafold # setup only needs to be run once
$ gget alphafold MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTFSYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK
</code></pre>
<p>Python (Jupyter Lab / Google Colab):</p>
<pre><code class="language-python">import gget
gget.ref(&quot;homo_sapiens&quot;)
gget.search([&quot;ace2&quot;, &quot;angiotensin converting enzyme 2&quot;], &quot;homo_sapiens&quot;)
gget.info([&quot;ENSG00000130234&quot;, &quot;ENST00000252519&quot;])
gget.seq(&quot;ENSG00000130234&quot;, translate=True)
gget.blat(&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;)
gget.blast(&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;)
gget.muscle(&quot;path/to/file.fa&quot;)
gget.enrichr([&quot;ACE2&quot;, &quot;AGT&quot;, &quot;AGTR1&quot;, &quot;ACE&quot;, &quot;AGTRAP&quot;, &quot;AGTR2&quot;, &quot;ACE3P&quot;], database=&quot;ontology&quot;, plot=True)
gget.archs4(&quot;ACE2&quot;, which=&quot;tissue&quot;)
gget.pdb(&quot;1R42&quot;, save=True)

gget.setup(&quot;alphafold&quot;) # setup only needs to be run once
gget.alphafold(&quot;MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTFSYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK&quot;)
</code></pre>
<p>Call <code>gget</code> from R using <a href="https://rstudio.github.io/reticulate/">reticulate</a>:</p>
<pre><code class="language-r">system(&quot;pip install gget&quot;)
install.packages(&quot;reticulate&quot;)
library(reticulate)
gget &lt;- import(&quot;gget&quot;)

gget$ref(&quot;homo_sapiens&quot;)
gget$search(list(&quot;ace2&quot;, &quot;angiotensin converting enzyme 2&quot;), &quot;homo_sapiens&quot;)
gget$info(list(&quot;ENSG00000130234&quot;, &quot;ENST00000252519&quot;))
gget$seq(&quot;ENSG00000130234&quot;, translate=TRUE)
gget$blat(&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;)
gget$blast(&quot;MSSSSWLLLSLVAVTAAQSTIEEQAKTFLDKFNHEAEDLFYQSSLAS&quot;)
gget$muscle(&quot;path/to/file.fa&quot;, out=&quot;path/to/out.afa&quot;)
gget$enrichr(list(&quot;ACE2&quot;, &quot;AGT&quot;, &quot;AGTR1&quot;, &quot;ACE&quot;, &quot;AGTRAP&quot;, &quot;AGTR2&quot;, &quot;ACE3P&quot;), database=&quot;ontology&quot;)
gget$archs4(&quot;ACE2&quot;, which=&quot;tissue&quot;)
gget$pdb(&quot;1R42&quot;, save=TRUE)
</code></pre>
<h4 id="more-examples"><a class="header" href="#more-examples"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-ref-"><a class="header" href="#gget-ref-">gget ref 📖</a></h2>
<p>Fetch FTPs and their respective metadata (or use flag <code>ftp</code> to only return the links) for reference genomes and annotations from <a href="https://www.ensembl.org/">Ensembl</a> by species.<br />
Return format: dictionary/JSON.</p>
<p><strong>Positional argument</strong><br />
<code>species</code><br />
Species for which the FTPs will be fetched in the format genus_species, e.g. homo_sapiens.<br />
Note: Not required when calling flag <code>--list_species</code>.<br />
Supported shortcuts: 'human', 'mouse'</p>
<p><strong>Optional arguments</strong><br />
<code>-w</code> <code>--which</code><br />
Defines which results to return. Default: 'all' -&gt; Returns all available results.<br />
Possible entries are one or a combination (as comma-separated list) of the following:<br />
'gtf' - Returns the annotation (GTF).<br />
'cdna' - Returns the trancriptome (cDNA).<br />
'dna' - Returns the genome (DNA).<br />
'cds' - Returns the coding sequences corresponding to Ensembl genes. (Does not contain UTR or intronic sequence.)<br />
'cdrna' - Returns transcript sequences corresponding to non-coding RNA genes (ncRNA).<br />
'pep' - Returns the protein translations of Ensembl genes.</p>
<p><code>-r</code> <code>--release</code><br />
Defines the Ensembl release number from which the files are fetched, e.g. 104. Default: latest Ensembl release.</p>
<p><code>-o</code> <code>--out</code><br />
Path to the JSON file the results will be saved in, e.g. path/to/directory/results.json. Default: Standard out.<br />
Python: <code>save=True</code> will save the output in the current working directory.</p>
<p><strong>Flags</strong><br />
<code>-l</code> <code>--list_species</code><br />
Lists all available species. (Python: combine with <code>species=None</code>.)</p>
<p><code>-ftp</code> <code>--ftp</code><br />
Returns only the requested FTP links.</p>
<p><code>-d</code> <code>--download</code><br />
Command-line only. Downloads the requested FTPs to the current directory (requires <a href="https://curl.se/docs/">curl</a> to be installed).</p>
<h3 id="examples"><a class="header" href="#examples">Examples</a></h3>
<p><strong>Use <code>gget ref</code> in combination with <a href="https://www.kallistobus.tools/kb_usage/kb_ref/">kallisto | bustools</a> to build a reference index:</strong></p>
<pre><code class="language-bash">kb ref -i INDEX -g T2G -f1 FASTA $(gget ref --ftp -w dna,gtf homo_sapiens)
</code></pre>
<p>→ kb ref builds a reference index using the latest DNA and GTF files of species <strong>Homo sapiens</strong> passed to it by <code>gget ref</code>.</p>
<p><strong>List all available genomes from Ensembl release 103:</strong></p>
<pre><code class="language-bash">gget ref --list_species -r 103
</code></pre>
<pre><code class="language-python"># Python
gget.ref(species=None, list_species=True, release=103)
</code></pre>
<p>→ Returns a list with all available genomes (checks if GTF and FASTAs are available) from Ensembl release 103.<br />
(If no release is specified, <code>gget ref</code> will always return information from the latest Ensembl release.)</p>
<p><strong>Get the genome reference for a specific species:</strong></p>
<pre><code class="language-bash">gget ref -w gtf,dna homo_sapiens
</code></pre>
<pre><code class="language-python"># Python
gget.ref(&quot;homo_sapiens&quot;, which=[&quot;gtf&quot;, &quot;dna&quot;])
</code></pre>
<p>→ Returns a JSON with the latest human GTF and FASTA FTPs, and their respective metadata, in the format:</p>
<pre><code>{
    &quot;homo_sapiens&quot;: {
        &quot;annotation_gtf&quot;: {
            &quot;ftp&quot;: &quot;http://ftp.ensembl.org/pub/release-106/gtf/homo_sapiens/Homo_sapiens.GRCh38.106.gtf.gz&quot;,
            &quot;ensembl_release&quot;: 106,
            &quot;release_date&quot;: &quot;28-Feb-2022&quot;,
            &quot;release_time&quot;: &quot;23:27&quot;,
            &quot;bytes&quot;: &quot;51379459&quot;
        },
        &quot;genome_dna&quot;: {
            &quot;ftp&quot;: &quot;http://ftp.ensembl.org/pub/release-106/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz&quot;,
            &quot;ensembl_release&quot;: 106,
            &quot;release_date&quot;: &quot;21-Feb-2022&quot;,
            &quot;release_time&quot;: &quot;09:35&quot;,
            &quot;bytes&quot;: &quot;881211416&quot;
        }
    }
}
</code></pre>
<h4 id="more-examples-1"><a class="header" href="#more-examples-1"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-search-"><a class="header" href="#gget-search-">gget search 🔎</a></h2>
<p>Fetch genes and transcripts from <a href="https://www.ensembl.org/">Ensembl</a> using free-form search terms.<br />
Return format: JSON (command-line) or data frame/CSV (Python).</p>
<p><strong>Positional argument</strong>
<code>searchwords</code><br />
One or more free form search words, e.g. gaba nmda. (Note: Search is not case-sensitive.)</p>
<p><strong>Other required arguments</strong><br />
<code>-s</code> <code>--species</code><br />
Species or database to be searched.<br />
A species can be passed in the format 'genus_species', e.g. 'homo_sapiens'.<br />
To pass a specific database, pass the name of the CORE database, e.g. 'mus_musculus_dba2j_core_105_1'.<br />
All availabale databases can be found <a href="http://ftp.ensembl.org/pub/release-106/mysql/">here</a>.<br />
Supported shortcuts: 'human', 'mouse'. </p>
<p><strong>Optional arguments</strong><br />
<code>-t</code> <code>--id_type</code><br />
'gene' (default) or 'transcript'<br />
Returns genes or transcripts, respectively.</p>
<p><code>-ao</code> <code>--andor</code><br />
'or' (default) or 'and'<br />
'or': Returns all genes that INCLUDE AT LEAST ONE of the searchwords in their name/description.<br />
'and': Returns only genes that INCLUDE ALL of the searchwords in their name/description.</p>
<p><code>-l</code> <code>--limit</code><br />
Limits the number of search results, e.g. 10. Default: None.</p>
<p><code>-o</code> <code>--out</code><br />
Path to the csv the results will be saved in, e.g. path/to/directory/results.csv (or .json). Default: Standard out.<br />
Python: <code>save=True</code> will save the output in the current working directory.</p>
<p><strong>Flags</strong><br />
<code>-csv</code> <code>--csv</code><br />
Command-line only. Returns results in CSV format.<br />
Python: Use <code>json=True</code> to return output in JSON format.</p>
<p><code>wrap_text</code><br />
Python only. <code>wrap_text=True</code> displays data frame with wrapped text for easy reading (default: False).</p>
<h3 id="example"><a class="header" href="#example">Example</a></h3>
<pre><code class="language-bash">gget search -s human gaba gamma-aminobutyric
</code></pre>
<pre><code class="language-python"># Python
gget.search([&quot;gaba&quot;, &quot;gamma-aminobutyric&quot;], &quot;homo_sapiens&quot;)
</code></pre>
<p>→ Returns all genes that contain at least one of the search words in their name or Ensembl/external reference description:</p>
<table><thead><tr><th>ensembl_id</th><th>gene_name</th><th>ensembl_description</th><th>ext_ref_description</th><th>biotype</th><th>url</th></tr></thead><tbody>
<tr><td>ENSG00000034713</td><td>GABARAPL2</td><td>GABA type A receptor associated protein like 2 [Source:HGNC Symbol;Acc:HGNC:13291]</td><td>GABA type A receptor associated protein like 2</td><td>protein_coding</td><td>https://uswest.ensembl.org/homo_sapiens/Gene/Summary?g=ENSG00000034713</td></tr>
<tr><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td></tr>
</tbody></table>
<h4 id="more-examples-2"><a class="header" href="#more-examples-2"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-info-"><a class="header" href="#gget-info-">gget info 💡</a></h2>
<p>Fetch extensive gene and transcript metadata from <a href="https://www.ensembl.org/">Ensembl</a>, <a href="https://www.uniprot.org/">UniProt</a>, and <a href="https://www.ncbi.nlm.nih.gov/">NCBI</a> using Ensembl IDs.<br />
Return format: JSON (command-line) or data frame/CSV (Python).</p>
<p><strong>Positional argument</strong><br />
<code>ens_ids</code><br />
One or more Ensembl IDs.</p>
<p><strong>Optional arguments</strong><br />
<code>-o</code> <code>--out</code><br />
Path to the file the results will be saved in, e.g. path/to/directory/results.csv (or .json). Default: Standard out.<br />
Python: <code>save=True</code> will save the output in the current working directory.</p>
<p><strong>Flags</strong><br />
<code>-eo</code> <code>--ensembl_only</code><br />
Only return results from Ensembl (excludes results from UniProt, NCBI and PDB, but runs significantly faster).</p>
<p><code>-q</code> <code>--quiet</code><br />
Command-line only. Prevents progress information from being displayed.<br />
Python: Use <code>verbose=False</code> to prevent progress information from being displayed.</p>
<p><code>-csv</code> <code>--csv</code><br />
Command-line only. Returns results in CSV format.<br />
Python: Use <code>json=True</code> to return output in JSON format.</p>
<p><code>wrap_text</code><br />
Python only. <code>wrap_text=True</code> displays data frame with wrapped text for easy reading (default: False).</p>
<h3 id="example-1"><a class="header" href="#example-1">Example</a></h3>
<pre><code class="language-bash">gget info ENSG00000034713 ENSG00000104853 ENSG00000170296
</code></pre>
<pre><code class="language-python"># Python
gget.info([&quot;ENSG00000034713&quot;, &quot;ENSG00000104853&quot;, &quot;ENSG00000170296&quot;])
</code></pre>
<p>→ Returns extensive information about each requested Ensembl ID:</p>
<table><thead><tr><th></th><th>uniprot_id</th><th>ncbi_gene_id</th><th>primary_gene_name</th><th>synonyms</th><th>protein_names</th><th>ensembl_description</th><th>uniprot_description</th><th>ncbi_description</th><th>biotype</th><th>canonical_transcript</th><th>...</th></tr></thead><tbody>
<tr><td>ENSG00000034713</td><td>P60520</td><td>11345</td><td>GABARAPL2</td><td>[ATG8, ATG8C, FLC3A, GABARAPL2, GATE-16, GATE16, GEF-2, GEF2]</td><td>Gamma-aminobutyric acid receptor-associated protein like 2 (GABA(A) receptor-associated protein-like 2)...</td><td>GABA type A receptor associated protein like 2 [Source:HGNC Symbol;Acc:HGNC:13291]</td><td>FUNCTION: Ubiquitin-like modifier involved in intra- Golgi traffic (By similarity). Modulates intra-Golgi transport through coupling between NSF activity and ...</td><td>Enables ubiquitin protein ligase binding activity. Involved in negative regulation of proteasomal protein catabolic process and protein...</td><td>protein_coding</td><td>ENST00000037243.7</td><td>...</td></tr>
<tr><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>...</td></tr>
</tbody></table>
<h4 id="more-examples-3"><a class="header" href="#more-examples-3"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-seq-"><a class="header" href="#gget-seq-">gget seq 🧬</a></h2>
<p>Fetch nucleotide or amino acid sequence of a gene (and all its isoforms) or a transcript by Ensembl ID.<br />
Return format: FASTA.</p>
<p><strong>Positional argument</strong><br />
<code>ens_ids</code><br />
One or more Ensembl IDs.</p>
<p><strong>Optional arguments</strong><br />
<code>-o</code> <code>--out</code><br />
Path to the file the results will be saved in, e.g. path/to/directory/results.fa. Default: Standard out.<br />
Python: <code>save=True</code> will save the output in the current working directory.</p>
<p><strong>Flags</strong><br />
<code>-t</code> <code>--translate</code><br />
Returns amino acid (instead of nucleotide) sequences.<br />
Nucleotide sequences are fetched from <a href="https://www.ensembl.org/">Ensembl</a>.<br />
Amino acid sequences are fetched from <a href="https://www.uniprot.org/">UniProt</a>.</p>
<p><code>-iso</code> <code>--isoforms</code><br />
Returns the sequences of all known transcripts.<br />
(Only for gene IDs.)</p>
<h3 id="examples-1"><a class="header" href="#examples-1">Examples</a></h3>
<pre><code class="language-bash">gget seq ENSG00000034713 ENSG00000104853 ENSG00000170296
</code></pre>
<pre><code class="language-python"># Python
gget.seq([&quot;ENSG00000034713&quot;, &quot;ENSG00000104853&quot;, &quot;ENSG00000170296&quot;])
</code></pre>
<p>→ Returns the nucleotide sequences of ENSG00000034713, ENSG00000104853, and ENSG00000170296 in FASTA format.</p>
<pre><code class="language-bash">gget seq -t -iso ENSG00000034713
</code></pre>
<pre><code class="language-python"># Python
gget.seq(&quot;ENSG00000034713&quot;, translate=True, isoforms=True)
</code></pre>
<p>→ Returns the amino acid sequences of all known transcripts of ENSG00000034713 in FASTA format.</p>
<h4 id="more-examples-4"><a class="header" href="#more-examples-4"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python.  The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-blast-"><a class="header" href="#gget-blast-">gget blast 💥</a></h2>
<p>BLAST a nucleotide or amino acid sequence to any <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a> database.<br />
Return format: JSON (command-line) or data frame/CSV (Python).</p>
<p><strong>Positional argument</strong><br />
<code>sequence</code><br />
Nucleotide or amino acid sequence, or path to FASTA or .txt file.</p>
<p><strong>Optional arguments</strong><br />
<code>-p</code> <code>--program</code><br />
'blastn', 'blastp', 'blastx', 'tblastn', or 'tblastx'.<br />
Default: 'blastn' for nucleotide sequences; 'blastp' for amino acid sequences.</p>
<p><code>-db</code> <code>--database</code><br />
'nt', 'nr', 'refseq_rna', 'refseq_protein', 'swissprot', 'pdbaa', or 'pdbnt'.<br />
Default: 'nt' for nucleotide sequences; 'nr' for amino acid sequences.<br />
<a href="https://ncbi.github.io/blast-cloud/blastdb/available-blastdbs.html">More info on BLAST databases</a></p>
<p><code>-l</code> <code>--limit</code><br />
Limits number of hits to return. Default: 50.</p>
<p><code>-e</code> <code>--expect</code><br />
Defines the <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=FAQ#expect">expect value</a> cutoff. Default: 10.0.</p>
<p><code>-o</code> <code>--out</code><br />
Path to the file the results will be saved in, e.g. path/to/directory/results.csv (or .json). Default: Standard out.<br />
Python: <code>save=True</code> will save the output in the current working directory.</p>
<p><strong>Flags</strong><br />
<code>-lcf</code> <code>--low_comp_filt</code><br />
Turns on <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=FAQ#LCR">low complexity filter</a>.</p>
<p><code>-mbo</code> <code>--megablast_off</code><br />
Turns off MegaBLAST algorithm. Default: MegaBLAST on (blastn only).</p>
<p><code>-q</code> <code>--quiet</code><br />
Command-line only. Prevents progress information from being displayed.<br />
Python: Use <code>verbose=False</code> to prevent progress information from being displayed.</p>
<p><code>-csv</code> <code>--csv</code><br />
Command-line only. Returns results in CSV format.<br />
Python: Use <code>json=True</code> to return output in JSON format.</p>
<p><code>wrap_text</code><br />
Python only. <code>wrap_text=True</code> displays data frame with wrapped text for easy reading (default: False).</p>
<h3 id="example-2"><a class="header" href="#example-2">Example</a></h3>
<pre><code class="language-bash">gget blast MKWMFKEDHSLEHRCVESAKIRAKYPDRVPVIVEKVSGSQIVDIDKRKYLVPSDITVAQFMWIIRKRIQLPSEKAIFLFVDKTVPQSR
</code></pre>
<pre><code class="language-python"># Python
gget.blast(&quot;MKWMFKEDHSLEHRCVESAKIRAKYPDRVPVIVEKVSGSQIVDIDKRKYLVPSDITVAQFMWIIRKRIQLPSEKAIFLFVDKTVPQSR&quot;)
</code></pre>
<p>→ Returns the BLAST result of the sequence of interest. <code>gget blast</code> automatically detects this sequence as an amino acid sequence and therefore sets the BLAST program to <em>blastp</em> with database <em>nr</em>.</p>
<table><thead><tr><th>Description</th><th>Scientific Name</th><th>Common Name</th><th>Taxid</th><th>Max Score</th><th>Total Score</th><th>Query Cover</th><th>...</th></tr></thead><tbody>
<tr><td>PREDICTED: gamma-aminobutyric acid receptor-as...</td><td>Colobus angolensis palliatus</td><td>NaN</td><td>336983</td><td>180</td><td>180</td><td>100%</td><td>...</td></tr>
<tr><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>...</td></tr>
</tbody></table>
<p><strong>BLAST from .fa or .txt file:</strong></p>
<pre><code class="language-bash">gget blast fasta.fa
</code></pre>
<pre><code class="language-python"># Python
gget.blast(&quot;fasta.fa&quot;)
</code></pre>
<p>→ Returns the BLAST results of the first sequence contained in the fasta.fa file. </p>
<h4 id="more-examples-5"><a class="header" href="#more-examples-5"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-blat-"><a class="header" href="#gget-blat-">gget blat 🎯</a></h2>
<p>Find the genomic location of a nucleotide or amino acid sequence using <a href="https://genome.ucsc.edu/cgi-bin/hgBlat">BLAT</a>.<br />
Return format: JSON (command-line) or data frame/CSV (Python).</p>
<p><strong>Positional argument</strong><br />
<code>sequence</code><br />
Nucleotide or amino acid sequence, or path to FASTA or .txt file.</p>
<p><strong>Optional arguments</strong><br />
<code>-st</code> <code>--seqtype</code><br />
'DNA', 'protein', 'translated%20RNA', or 'translated%20DNA'.<br />
Default: 'DNA' for nucleotide sequences; 'protein' for amino acid sequences.</p>
<p><code>-a</code> <code>--assembly</code><br />
'human' (hg38) (default), 'mouse' (mm39), 'zebrafinch' (taeGut2),<br />
or any of the species assemblies available <a href="https://genome.ucsc.edu/cgi-bin/hgBlat">here</a> (use short assembly name).</p>
<p><code>-o</code> <code>--out</code><br />
Path to the file the results will be saved in, e.g. path/to/directory/results.csv (or .json). Default: Standard out.<br />
Python: <code>save=True</code> will save the output in the current working directory.</p>
<p><strong>Flags</strong><br />
<code>-csv</code> <code>--csv</code><br />
Command-line only. Returns results in CSV format.<br />
Python: Use <code>json=True</code> to return output in JSON format.</p>
<h3 id="example-3"><a class="header" href="#example-3">Example</a></h3>
<pre><code class="language-bash">gget blat -a taeGut2 MKWMFKEDHSLEHRCVESAKIRAKYPDRVPVIVEKVSGSQIVDIDKRKYLVPSDITVAQFMWIIRKRIQLPSEKAIFLFVDKTVPQSR
</code></pre>
<pre><code class="language-python"># Python
gget.blat(&quot;MKWMFKEDHSLEHRCVESAKIRAKYPDRVPVIVEKVSGSQIVDIDKRKYLVPSDITVAQFMWIIRKRIQLPSEKAIFLFVDKTVPQSR&quot;, assembly=&quot;taeGut2&quot;)
</code></pre>
<p>→ Returns BLAT results for assembly taeGut2 (zebra finch). In the above example, <code>gget blat</code> automatically detects this sequence as an amino acid sequence and therefore sets the BLAT seqtype to <em>protein</em>.</p>
<table><thead><tr><th>genome</th><th>query_size</th><th>aligned_start</th><th>aligned_end</th><th>matches</th><th>mismatches</th><th>%_aligned</th><th>...</th></tr></thead><tbody>
<tr><td>taeGut2</td><td>88</td><td>12</td><td>88</td><td>77</td><td>0</td><td>87.5</td><td>...</td></tr>
</tbody></table>
<h4 id="more-examples-6"><a class="header" href="#more-examples-6"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-muscle-"><a class="header" href="#gget-muscle-">gget muscle 🦾</a></h2>
<p>Align multiple nucleotide or amino acid sequences to each other using <a href="https://www.drive5.com/muscle/">Muscle5</a>.<br />
Return format: ClustalW formatted standard out or aligned FASTA (.afa).</p>
<p><strong>Positional argument</strong><br />
<code>fasta</code><br />
Path to FASTA or .txt file containing the nucleotide or amino acid sequences to be aligned.</p>
<p><strong>Optional arguments</strong><br />
<code>-o</code> <code>--out</code><br />
Path to the aligned FASTA file the results will be saved in, e.g. path/to/directory/results.afa. Default: Standard out.<br />
Python: <code>save=True</code> will save the output in the current working directory.</p>
<p><strong>Flags</strong><br />
<code>-s5</code> <code>--super5</code><br />
Aligns input using the <a href="https://drive5.com/muscle5/Muscle5_SuppMat.pdf">Super5 algorithm</a> instead of the <a href="https://drive5.com/muscle5/Muscle5_SuppMat.pdf">Parallel Perturbed Probcons (PPP) algorithm</a> to decrease time and memory.<br />
Use for large inputs (a few hundred sequences).</p>
<h3 id="example-4"><a class="header" href="#example-4">Example</a></h3>
<pre><code class="language-bash">gget muscle fasta.fa
</code></pre>
<pre><code class="language-python"># Python
gget.muscle(&quot;fasta.fa&quot;)
</code></pre>
<p>→ Returns an overview of the aligned sequences with ClustalW coloring. (To return an aligned FASTA (.afa) file, use <code>--out</code> argument (or <code>save=True</code> in Jupyter Lab/Google Colab).) In the above example, the 'fasta.fa' includes several sequences to be aligned (e.g. isoforms returned from <code>gget seq</code>). </p>
<p><img src="https://github.com/pachterlab/gget/blob/main/figures/example_muscle_return.png?raw=true" alt="alt text" /></p>
<h4 id="more-examples-7"><a class="header" href="#more-examples-7"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-enrichr-"><a class="header" href="#gget-enrichr-">gget enrichr 💰</a></h2>
<p>Perform an enrichment analysis on a list of genes using <a href="https://maayanlab.cloud/Enrichr/">Enrichr</a>.<br />
Return format: JSON (command-line) or data frame/CSV (Python).</p>
<p><strong>Positional argument</strong><br />
<code>genes</code><br />
Short names (gene symbols) of genes to perform enrichment analysis on, e.g. PHF14 RBM3 MSL1 PHF21A.<br />
Alternatively: use flag <code>--ensembl</code> to input a list of Ensembl gene IDs, e.g. ENSG00000106443 ENSG00000102317 ENSG00000188895.</p>
<p><strong>Other required arguments</strong><br />
<code>-db</code> <code>--database</code><br />
Database to use as reference for the enrichment analysis.<br />
Supports any database listed <a href="https://maayanlab.cloud/Enrichr/#libraries">here</a> under 'Gene-set Library' or one of the following shortcuts:<br />
'pathway'       (KEGG_2021_Human)<br />
'transcription'     (ChEA_2016)<br />
'ontology'      (GO_Biological_Process_2021)<br />
'diseases_drugs'   (GWAS_Catalog_2019)<br />
'celltypes'      (PanglaoDB_Augmented_2021)<br />
'kinase_interactions'   (KEA_2015)</p>
<p><strong>Optional arguments</strong><br />
<code>-o</code> <code>--out</code><br />
Path to the file the results will be saved in, e.g. path/to/directory/results.csv (or .json). Default: Standard out.<br />
Python: <code>save=True</code> will save the output in the current working directory.</p>
<p><code>figsize</code><br />
Python only. (width, height) of plot in inches. (Default: (10,10))</p>
<p><code>ax</code><br />
Python only. Pass a matplotlib axes object for plot customization. (Default: None)</p>
<p><strong>Flags</strong><br />
<code>-e</code> <code>--ensembl</code><br />
Add this flag if <code>genes</code> are given as Ensembl gene IDs.</p>
<p><code>-csv</code> <code>--csv</code><br />
Command-line only. Returns results in CSV format.<br />
Python: Use <code>json=True</code> to return output in JSON format.</p>
<p><code>plot</code><br />
Python only. <code>plot=True</code> provides a graphical overview of the first 15 results (default: False).</p>
<h3 id="example-5"><a class="header" href="#example-5">Example</a></h3>
<pre><code class="language-bash">gget enrichr -db ontology ACE2 AGT AGTR1
</code></pre>
<pre><code class="language-python"># Python
gget.enrichr([&quot;ACE2&quot;, &quot;AGT&quot;, &quot;AGTR1&quot;], database=&quot;ontology&quot;, plot=True)
</code></pre>
<p>→ Returns pathways/functions involving genes ACE2, AGT, and AGTR1 from the <em>GO Biological Process 2021</em> database. In Python, <code>plot=True</code> returns a graphical overview of the results:</p>
<p><img src="https://github.com/pachterlab/gget/blob/main/figures/gget_enrichr_results.png?raw=true" alt="alt text" /></p>
<h4 id="more-examples-8"><a class="header" href="#more-examples-8"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-archs4-"><a class="header" href="#gget-archs4-">gget archs4 🐁</a></h2>
<p>Find the most correlated genes to a gene of interest or find the gene's tissue expression atlas using <a href="https://maayanlab.cloud/archs4/">ARCHS4</a>.<br />
Return format: JSON (command-line) or data frame/CSV (Python).</p>
<p><strong>Positional argument</strong><br />
<code>gene</code><br />
Short name (gene symbol) of gene of interest, e.g. STAT4.<br />
Alternatively: use flag <code>--ensembl</code> to input an Ensembl gene IDs, e.g. ENSG00000138378.</p>
<p><strong>Optional arguments</strong><br />
<code>-w</code> <code>--which</code><br />
'correlation' (default) or 'tissue'.<br />
'correlation' returns a gene correlation table that contains the 100 most correlated genes to the gene of interest. The Pearson correlation is calculated over all samples and tissues in <a href="https://maayanlab.cloud/archs4/">ARCHS4</a>.<br />
'tissue' returns a tissue expression atlas calculated from human or mouse samples (as defined by 'species') in <a href="https://maayanlab.cloud/archs4/">ARCHS4</a>.</p>
<p><code>-s</code> <code>--species</code><br />
'human' (default) or 'mouse'.<br />
Defines whether to use human or mouse samples from <a href="https://maayanlab.cloud/archs4/">ARCHS4</a>.<br />
(Only for tissue expression atlas.)</p>
<p><code>-o</code> <code>--out</code><br />
Path to the file the results will be saved in, e.g. path/to/directory/results.csv (or .json). Default: Standard out.<br />
Python: <code>save=True</code> will save the output in the current working directory.</p>
<p><strong>Flags</strong><br />
<code>-e</code> <code>--ensembl</code><br />
Add this flag if <code>gene</code> is given as an Ensembl gene ID.</p>
<p><code>-csv</code> <code>--csv</code><br />
Command-line only. Returns results in CSV format.<br />
Python: Use <code>json=True</code> to return output in JSON format.</p>
<h3 id="examples-2"><a class="header" href="#examples-2">Examples</a></h3>
<pre><code class="language-bash">gget archs4 ACE2
</code></pre>
<pre><code class="language-python"># Python
gget.archs4(&quot;ACE2&quot;)
</code></pre>
<p>→ Returns the 100 most correlated genes to ACE2:</p>
<table><thead><tr><th>gene_symbol</th><th>pearson_correlation</th></tr></thead><tbody>
<tr><td>SLC5A1</td><td>0.579634</td></tr>
<tr><td>CYP2C18</td><td>0.576577</td></tr>
<tr><td>. . .</td><td>. . .</td></tr>
</tbody></table>
<pre><code class="language-bash">gget archs4 -w tissue ACE2
</code></pre>
<pre><code class="language-python"># Python
gget.archs4(&quot;ACE2&quot;, which=&quot;tissue&quot;)
</code></pre>
<p>→ Returns the tissue expression of ACE2 (by default, human data is used):</p>
<table><thead><tr><th>id</th><th>min</th><th>q1</th><th>median</th><th>q3</th><th>max</th></tr></thead><tbody>
<tr><td>System.Urogenital/Reproductive System.Kidney.RENAL CORTEX</td><td>0.113644</td><td>8.274060</td><td>9.695840</td><td>10.51670</td><td>11.21970</td></tr>
<tr><td>System.Digestive System.Intestine.INTESTINAL EPITHELIAL CELL</td><td>0.113644</td><td>5.905560</td><td>9.570450</td><td>13.26470</td><td>13.83590</td></tr>
<tr><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td><td>. . .</td></tr>
</tbody></table>
<h4 id="more-examples-9"><a class="header" href="#more-examples-9"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-pdb-"><a class="header" href="#gget-pdb-">gget pdb 🔮</a></h2>
<p>Query <a href="https://www.rcsb.org/">RCSB Protein Data Bank (PDB)</a> for the protein structure/metadata of a given PDB ID.<br />
Return format: Resource 'pdb' is returned in PDB format. All other resources are returned in JSON format.</p>
<p><strong>Positional argument</strong><br />
<code>pdb_id</code><br />
PDB ID to be queried, e.g. '7S7U'.</p>
<p><strong>Optional arguments</strong><br />
<code>-r</code> <code>--resource</code><br />
Defines type of information to be returned. One of the following:<br />
'pdb': Returns the protein structure in PDB format (default).<br />
'entry': Information about PDB structures at the top level of PDB structure hierarchical data organization.<br />
'pubmed': Get PubMed annotations (data integrated from PubMed) for a given entry's primary citation.<br />
'assembly': Information about PDB structures at the quaternary structure level.<br />
'branched_entity': Get branched entity description (define entity ID as 'identifier').<br />
'nonpolymer_entity': Get non-polymer entity data (define entity ID as 'identifier').<br />
'polymer_entity': Get polymer entity data (define entity ID as 'identifier').<br />
'uniprot': Get UniProt annotations for a given macromolecular entity (define entity ID as 'identifier').<br />
'branched_entity_instance': Get branched entity instance description (define chain ID as 'identifier').<br />
'polymer_entity_instance': Get polymer entity instance (a.k.a chain) data (define chain ID as 'identifier').<br />
'nonpolymer_entity_instance': Get non-polymer entity instance description (define chain ID as 'identifier').</p>
<p><code>-i</code> <code>--identifier</code><br />
Can be used to define assembly, entity or chain ID (default: None). Assembly/entity IDs are numbers (e.g. 1), and chain IDs are letters (e.g. 'A').</p>
<p><code>-o</code> <code>--out</code><br />
Path to the file the results will be saved in, e.g. path/to/directory/7S7U.pdb or path/to/directory/7S7U_entry.json. Default: Standard out.<br />
Python: <code>save=True</code> will save the output in the current working directory.</p>
<h3 id="examples-3"><a class="header" href="#examples-3">Examples</a></h3>
<pre><code class="language-bash">gget pdb 7S7U -o 7S7U.pdb
</code></pre>
<pre><code class="language-python"># Python
gget.pdb(&quot;7S7U&quot;, save=True)
</code></pre>
<p>→ Saves the structure of 7S7U in PDB format as '7S7U.pdb' in the current working directory.</p>
<p><strong>Find PDB crystal structures for a comparative analysis of protein structure:</strong></p>
<pre><code class="language-bash"># Find PDB IDs associated with an Ensembl ID 
gget info ENSG00000130234

# Alternatively: Since many entries in the PDB do not have linked Ensembl IDs, 
# you will likely find more PDB entries by BLASTing the sequence agains the PDB.

# Get the amino acid sequence of a transcript from an Ensembl ID
gget seq --translate ENSG00000130234 -o gget_seq_results.fa

# BLAST an amino acid sequence to find similar structures in the PDB
gget blast --database pdbaa gget_seq_results.fa

# Get PDB files from the PDB IDs returned by gget blast for comparative analysis
gget pdb 7DQA -o 7DQA.pdb
gget pdb 7CT5 -o 7CT5.pdb
</code></pre>
<pre><code class="language-python"># Find PDB IDs associated with an Ensembl ID 
gget.info(&quot;ENSG00000130234&quot;)

# Alternatively: Since many entries in the PDB do not have linked Ensembl IDs, 
# you will likely find more PDB entries by BLASTing the sequence agains the PDB.

# Get the amino acid sequence of a transcript from an Ensembl ID
gget.seq(&quot;ENSG00000130234&quot;, translate=True, save=True)

# BLAST an amino acid sequence to find similar structures in the PDB
gget.blast(&quot;gget_seq_results.fa&quot;, database=&quot;pdbaa&quot;)

# Get PDB files from the PDB IDs returned by gget blast for comparative analysis
gget.pdb(&quot;7DQA&quot;, save=True)
gget.pdb(&quot;7CT5&quot;, save=True)
</code></pre>
<p>→ The use case above exemplifies how to find PDB files for comparative analysis of protein structure starting with Ensembl IDs or amino acid sequences. The fetched PDB files can also be compared to predicted structures generated by <a href="alpahfold.html"><code>gget alphafold</code></a>. PDB files can be viewed interactively in 3D <a href="https://rcsb.org/3d-view">online</a>, or using programs like <a href="https://pymol.org/">PyMOL</a> or <a href="https://www.blender.org/">Blender</a>. To compare two PDB files, you can use <a href="https://rcsb.org/alignment">this website</a>.</p>
<h4 id="more-examples-10"><a class="header" href="#more-examples-10"><a href="https://github.com/pachterlab/gget_examples">More examples</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-alphafold-"><a class="header" href="#gget-alphafold-">gget alphafold 🪢</a></h2>
<p>Predict the 3D structure of a protein from its amino acid sequence using a simplified version of <a href="https://www.deepmind.com/">DeepMind</a>’s <a href="https://github.com/deepmind/alphafold">AlphaFold2</a> originally released and benchmarked for <a href="https://colab.research.google.com/github/deepmind/alphafold/blob/main/notebooks/AlphaFold.ipynb">AlphaFold Colab</a>.<br />
Returns: Predicted structure (PDB) and alignment error (json).</p>
<p>Before using <code>gget alphafold</code> for the first time, run <code>gget setup alphafold</code> / <code>gget.setup(&quot;alphafold&quot;)</code> once (also see <a href="setup.html"><code>gget setup</code></a>).</p>
<p><strong>Positional argument</strong><br />
<code>sequence</code><br />
Amino acid sequence (str), list of sequences (for multimers), or path to FASTA file.</p>
<p><strong>Optional arguments</strong><br />
<code>-mr</code> <code>--multimer_recycles</code><br />
The multimer model will continue recycling until the predictions stop changing, up to the limit set here. Default: 3.<br />
For higher accuracy, at the potential cost of longer inference times, set this to 20.</p>
<p><code>-o</code> <code>--out</code><br />
Path to folder to save prediction results in (str). Default: &quot;./[date_time]_gget_alphafold_prediction&quot;.</p>
<p><strong>Flags</strong><br />
<code>-mfm</code> <code>--multimer_for_monomer</code><br />
Use multimer model for a monomer.</p>
<p><code>-r</code> <code>--relax</code><br />
AMBER relax the best model. </p>
<p><code>plot</code><br />
Python only. <code>plot=True</code> provides an interactive, 3D graphical overview of the predicted structure and alignment quality using <a href="https://pypi.org/project/py3Dmol/">py3Dmol</a> and <a href="https://matplotlib.org/">matplotlib</a> (default: True).</p>
<p><code>show_sidechains</code><br />
Python only. <code>show_sidechains=True</code> includes side chains in the plot (default: True).</p>
<h3 id="example-6"><a class="header" href="#example-6">Example</a></h3>
<pre><code class="language-bash"># Generate new prediction from amino acid sequence
gget alphafold MAAHKGAEHHHKAAEHHEQAAKHHHAAAEHHEKGEHEQAAHHADTAYAHHKHAEEHAAQAAKHDAEHHAPKPH

# Find similar sequences deposited on the PDB for comparative analysis
gget blast --database pdbaa MAAHKGAEHHHKAAEHHEQAAKHHHAAAEHHEKGEHEQAAHHADTAYAHHKHAEEHAAQAAKHDAEHHAPKPH

# Fetch the PDB files of similar structures returned by gget blast for comparison, to get a measure for model quality
gget pdb 3UQ3 -o 3UQ3.pdb
gget pdb 2K42 -o 2K42.pdb
</code></pre>
<pre><code class="language-python"># Python
gget.alphafold(&quot;MAAHKGAEHHHKAAEHHEQAAKHHHAAAEHHEKGEHEQAAHHADTAYAHHKHAEEHAAQAAKHDAEHHAPKPH&quot;)

# Find similar sequences deposited on the PDB for comparative analysis
gget.blast(&quot;MAAHKGAEHHHKAAEHHEQAAKHHHAAAEHHEKGEHEQAAHHADTAYAHHKHAEEHAAQAAKHDAEHHAPKPH&quot;, database=&quot;pdbaa&quot;)

# Fetch the PDB files of similar structures returned by gget blast for comparison, to get a measure for model quality
gget.pdb(&quot;3UQ3&quot;, save=True)
gget.pdb(&quot;2K42&quot;, save=True)
</code></pre>
<p>→ <code>gget alphafold</code> returns the predicted structure (PDB) and predicted alignment error (.json) in a new folder (&quot;./[date_time]_gget_alphafold_prediction&quot;). The use case above exemplifies how to use <a href="blast.html"><code>gget blast</code></a> and <a href="pdb.html"><code>gget pdb</code></a> for a comparative analysis of the new prediction. PDB files can be viewed interactively in 3D <a href="https://rcsb.org/3d-view">online</a>, or using programs like <a href="https://pymol.org/">PyMOL</a> or <a href="https://www.blender.org/">Blender</a>. To compare two PDB files, you can use <a href="https://rcsb.org/alignment">this website</a>. The Python interface also returns <a href="https://twitter.com/NeuroLuebbert/status/1555968042948915200">interactive plots</a>, which can be generated from the PDB and JSON as described in the <a href="https://github.com/pachterlab/gget/discussions/39">gget alphafold FAQ</a> Q4.</p>
<h4 id="example-in-google-colab"><a class="header" href="#example-in-google-colab"><a href="https://github.com/pachterlab/gget_examples/blob/main/gget_alphafold.ipynb">Example in Google Colab</a></a></h4>
<h4 id="also-see-gget-alphafold-faq"><a class="header" href="#also-see-gget-alphafold-faq">Also see: <a href="https://github.com/pachterlab/gget/discussions/39">gget alphafold FAQ</a></a></h4>
<div style="break-before: page; page-break-before: always;"></div><blockquote>
<p>Python arguments are equivalent to long-option arguments (<code>--arg</code>), unless otherwise specified. Flags are True/False arguments in Python. The manual for any gget tool can be called from the command-line using the <code>-h</code> <code>--help</code> flag.</p>
</blockquote>
<h2 id="gget-setup-"><a class="header" href="#gget-setup-">gget setup 🔧</a></h2>
<p>Function to install/download third-party dependencies for a specified gget module.</p>
<p><strong>Positional argument</strong><br />
<code>module</code><br />
gget module for which dependencies should be installed (currently only &quot;alphafold&quot;).</p>
<h3 id="example-7"><a class="header" href="#example-7">Example</a></h3>
<pre><code class="language-bash">gget setup alphafold
</code></pre>
<pre><code class="language-python"># Python
gget.setup(&quot;alphafold&quot;)
</code></pre>
<p>→ Installs all (modified) third-party dependencies and downloads model parameters (~4GB) required to run <code>gget alphafold</code>. </p>
<div style="break-before: page; page-break-before: always;"></div><p><a href="LICENSE"><img src="https://img.shields.io/pypi/l/gget" alt="license" /></a></p>
<h1 id="citation"><a class="header" href="#citation">Citation</a></h1>
<p>If you use <code>gget</code> in a publication, please cite:<br />
Luebbert, L., &amp; Pachter, L. (2023). Efficient querying of genomic reference databases with gget. Bioinformatics. https://doi.org/10.1093/bioinformatics/btac836</p>
<ul>
<li>
<p>If using <code>gget archs4</code>, please also cite:</p>
<ul>
<li>
<p>Lachmann A, Torre D, Keenan AB, Jagodnik KM, Lee HJ, Wang L, Silverstein MC, Ma’ayan A. Massive mining of publicly available RNA-seq data from human and mouse. Nature Communications 9. Article number: 1366 (2018), doi:10.1038/s41467-018-03751-6</p>
</li>
<li>
<p>Bray NL, Pimentel H, Melsted P and Pachter L, Near optimal probabilistic RNA-seq quantification, Nature Biotechnology 34, p 525--527 (2016). https://doi.org/10.1038/nbt.3519</p>
</li>
</ul>
</li>
<li>
<p>If using <code>gget enrichr</code>, please also cite:</p>
<ul>
<li>
<p>Chen EY, Tan CM, Kou Y, Duan Q, Wang Z, Meirelles GV, Clark NR, Ma'ayan A.
Enrichr: interactive and collaborative HTML5 gene list enrichment analysis tool. BMC Bioinformatics. 2013; 128(14).</p>
</li>
<li>
<p>Kuleshov MV, Jones MR, Rouillard AD, Fernandez NF, Duan Q, Wang Z, Koplev S, Jenkins SL, Jagodnik KM, Lachmann A, McDermott MG, Monteiro CD, Gundersen GW, Ma'ayan A.
Enrichr: a comprehensive gene set enrichment analysis web server 2016 update. Nucleic Acids Research. 2016; gkw377.</p>
</li>
<li>
<p>Xie Z, Bailey A, Kuleshov MV, Clarke DJB., Evangelista JE, Jenkins SL, Lachmann A, Wojciechowicz ML, Kropiwnicki E, Jagodnik KM, Jeon M, &amp; Ma’ayan A.
Gene set knowledge discovery with Enrichr. Current Protocols, 1, e90. 2021. doi: 10.1002/cpz1.90. </p>
</li>
</ul>
</li>
<li>
<p>If using <code>gget blast</code>, please also cite:</p>
<ul>
<li>Altschul SF, Gish W, Miller W, Myers EW, Lipman DJ. Basic local alignment search tool. J Mol Biol. 1990 Oct 5;215(3):403-10. doi: 10.1016/S0022-2836(05)80360-2. PMID: 2231712.</li>
</ul>
</li>
<li>
<p>If using <code>gget blat</code>, please also cite:</p>
<ul>
<li>Kent WJ. BLAT--the BLAST-like alignment tool. Genome Res. 2002 Apr;12(4):656-64. doi: 10.1101/gr.229202. PMID: 11932250; PMCID: PMC187518.</li>
</ul>
</li>
<li>
<p>If using <code>gget muscle</code>, please also cite:</p>
<ul>
<li>Edgar RC (2021), MUSCLE v5 enables improved estimates of phylogenetic tree confidence by ensemble bootstrapping, bioRxiv 2021.06.20.449169. https://doi.org/10.1101/2021.06.20.449169.</li>
</ul>
</li>
<li>
<p>If using <code>gget alphafold</code>, please also cite:</p>
<ul>
<li>Jumper, J., Evans, R., Pritzel, A. et al. Highly accurate protein structure prediction with AlphaFold. Nature 596, 583–589 (2021). https://doi.org/10.1038/s41586-021-03819-2</li>
</ul>
<p>And, if applicable:</p>
<ul>
<li>Evans, R. et al. Protein complex prediction with AlphaFold-Multimer. bioRxiv 2021.10.04.463034; https://doi.org/10.1101/2021.10.04.463034</li>
</ul>
</li>
</ul>
<hr />
<h1 id="disclaimer"><a class="header" href="#disclaimer">Disclaimer</a></h1>
<p><code>gget</code> is only as accurate as the databases/servers/APIs it queries from. The accuracy or reliability of the data is not guaranteed or warranted in any way and the providers disclaim liability of any kind whatsoever, including, without limitation, liability for quality, performance, merchantability and fitness for a particular purpose arising out of the use, or inability to use the data.</p>

                    </main>

                    <nav class="nav-wrapper" aria-label="Page navigation">
                        <!-- Mobile navigation buttons -->
                        <div style="clear: both"></div>
                    </nav>
                </div>
            </div>

            <nav class="nav-wide-wrapper" aria-label="Page navigation">
            </nav>

        </div>

        <script type="text/javascript">
            window.playground_copyable = true;
        </script>
        <script src="elasticlunr.min.js" type="text/javascript" charset="utf-8"></script>
        <script src="mark.min.js" type="text/javascript" charset="utf-8"></script>
        <script src="searcher.js" type="text/javascript" charset="utf-8"></script>
        <script src="clipboard.min.js" type="text/javascript" charset="utf-8"></script>
        <script src="highlight.js" type="text/javascript" charset="utf-8"></script>
        <script src="book.js" type="text/javascript" charset="utf-8"></script>

        <!-- Custom JS scripts -->
        <script type="text/javascript">
        window.addEventListener('load', function() {
            window.setTimeout(window.print, 100);
        });
        </script>
    </body>
</html>
